code/modules/assembly/flash.dm

```diff
@@ -71,11 +71,13 @@
 /obj/item/device/assembly/flash/proc/AOE_flash(bypass_checks = FALSE, range = 3, power = 5, targeted = FALSE, mob/user)
 	if(!bypass_checks && !try_use_flash())
 		return FALSE
-	var/list/mob/targets = get_flash_targets(get_turf(src), range, FALSE)
+  
+	var/list/targets = get_flash_targets(get_turf(src), range, FALSE)
 	if(user)
 		targets -= user
-	for(var/mob/living/carbon/C in targets)
-		flash_carbon(C, user, power, targeted, TRUE)
+	for(var/i in targets)
+		var/mob/living/L= i
+		flash_carbon(L, user, power, targeted, TRUE)

```

Comment:
But this istype is ran on flash_carbon regardless?

I would imagine 2 typechecks (both in the loop as well as flash_carbon) > 1 typecheck in just flash_carbon?

Reply:
Filtered lists aren't an istype, they're cheaper. It also avoids the proc call overhead.

Reply:
From Contributing.md

>The above is a simple proc for checking all swords in a container and returning the one with the highest damage, and it uses DM's standard syntax for a for-loop by specifying a type in the variable of the for's header that DM interprets as a type to filter by. It performs this filter using istype() (or some internal-magic similar to istype() - this is BYOND, after all). This is fine in its current state for bag_of_items, but if bag_of_items contained ONLY swords, or only SUBTYPES of swords, then the above is inefficient. For example:

Reply:
If we updated the contributing guidelines to reflect every performance oddity of byond it'd by mostly notes on performance. This isn't important enough information to merit putting it in there imo. Either way, that'd be another pr. Ask about it in #coderbus

Reply:
this is some bullshit, we dont' even know which way is faster because neither of you tested it.

since OP is the one claiming to have "micro optimisations" the onus is on you to provide benchmarks.

Otherwise we're just pissing into the wind

Reply:
^ that basically

Various optimizations that save small amounts of time change in signature all the time in different releases of byond. Filtered loops in particular got a performance boost in the last few months that wasn't noted in changelogs but changed which type of loop was faster in certain situations.

All we can tell you is which method is most likely to be faster.